pkg/ingester/wal.go

```diff
@@ -569,6 +633,35 @@ type samplesWithUserID struct {
 	userID  string
 }
 
+func processWALWithRepair(walDir string, startSegment int, userStates *userStates, nWorkers int,
```

Comment:
Would it make sense to bundle all these parameters into some struct, and make these functions methods on the struct? We seem to be passing them around to different methods.

Reply:
Done